### PR TITLE
Fix + improve automated rendering

### PR DESCRIPTION
**Pull request by aviflax**
Simon Brown, the owner and maintainer of Structurizr Express, has made some changes to its code that broke our renderer.

Thankfully, he also published [a new example script](https://github.com/structurizr/puppeteer) that demonstrates the optimal way to render and export diagrams via Structurizr Express; this enabled me to both fix our renderer but also improve it.

So this adapts the renderer to the changes but also makes our code simpler and faster.

This required no test changes; the (Clojure) test that invokes the renderer is passing.

### CODE REVIEW

File: tool/renderer/render.js [hunk 1]
```diff
@@ -142,20 +96,29 @@ async function closeBrowser(browser, debugMode) {
...
   log.next('launching browser');
   const opts = puppeteerOpts(debugMode);
   const browser = await puppeteer.launch(opts);
 
-  const imageBuffer = await render(theYaml, browser, url, debugMode);
+  const imageBuffer = await render(preppedYaml, browser, url, debugMode);
   closeBrowser(browser, debugMode);
 
   process.stdout.write(imageBuffer);
 }
```
Comment (fussybeaver) [inline tool/renderer/render.js]: should there be some error handling somewhere? There's a lot of file operations, opening browsers, presumably in a different process. Not sure what is likely to fail here - is there any other sanitization done on the arguments to the main method ?

File: tool/renderer/render.js [hunk 2]
```diff
@@ -50,86 +50,40 @@ function puppeteerOpts(debugMode) {
...
-  const newPagePromise = new Promise(resolve => {
-    mainPage.browser().once('targetcreated', target => resolve(target.page()));
-  });
-
-  await abit(200);
-  await mainPage.evaluate(() => {
-    Structurizr.diagram.exportCurrentView(1, true, false, false, false);
+async function exportDiagram(page) {
+  const diagramImageBase64DataURI = await page.evaluate(() => {
+    return structurizr.scripting.exportCurrentDiagramToPNG();
```
Comment (fussybeaver) [inline tool/renderer/render.js]: This doesn't actually do any manipulation of the image data in javascript ?

Comment (aviflax) [reply tool/renderer/render.js]: Correct. The purpose of the renderer — well honestly it’s not _really_ a renderer! Structurizr Express is doing the rendering; this script is more of an adapter that uses Structurizr Express to render YAML files to PNG files. Maybe I should rename the script?

File: tool/renderer/render.js (hunk 1 above)
Comment (aviflax) [reply tool/renderer/render.js]: Yes, there should definitely be error handling. I do need to add it ASAP — I’ve created #99 to track that. Thank you!

**Pull request merged**